src/index.ts

```diff
@@ -10,7 +10,10 @@ export type Options = {
 		samePage: boolean;
 	};
 	getAnchorElement?: (hash: string) => Element | null;
-	offset: number | ((scrollTarget?: Element, scrollContainer?: Element) => number);
+	offset:
+		| number
+		| ScrollPosition
+		| ((scrollTarget?: Element, scrollContainer?: Element) => number | ScrollPosition);
```

Comment:
I just stumbled upon this: Are `scrollTarget` and `scrollContainer` not always defined? `getOffset` is only being called by `scrollElementIntoView()`, where both are always defined.

Reply:
I'm currently confused – is the offset to be applied when doing a normal scroll to top (without a `scrollTarget`)? If so, `getOffset` should be changed to this:

```js
((scrollContainer: Element, scrollTarget?: Element) => number | ScrollPosition)
```

Reply:
True, these should always be defined. We can probably safely change the signature.

Reply:
The offset is only applied when scrolling to an element. When scrolling to the top, it wouldn't have an effect. Subtracting the offset would end up with a scroll position below zero, so still technically the top.